src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -816,7 +844,40 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                 Debug.Assert(idx == utf8Text.Length);
 
                 idx = -1; // All bytes are allowed.
+                goto Return;
 
+#if NETCOREAPP
+            Return15:
+                return idx + 15;
+            Return14:
+                return idx + 14;
+            Return13:
+                return idx + 13;
+            Return12:
+                return idx + 12;
+            Return11:
+                return idx + 11;
+            Return10:
+                return idx + 10;
+            Return9:
+                return idx + 9;
+            Return8:
+                return idx + 8;
+            Return7:
+                return idx + 7;
+            Return6:
+                return idx + 6;
+            Return5:
+                return idx + 5;
+            Return4:
+                return idx + 4;
+            Return3:
+                return idx + 3;
+            Return2:
+                return idx + 2;
+            Return1:
+                return idx + 1;
+#endif
             Return:
```

Comment:
Maybe move this label before the `#if NETCOREAPP`, so that in the SSSE3 path, the jump is closer to the rest of the code?

Reply:
Good point. Fortunately the JIT is smart enough to remove the labels that are not reachable.